lib/src/base_impl.dart

```diff
@@ -9,83 +9,41 @@ import 'dart:convert';
 import 'exceptions.dart';
 import 'reddit.dart';
 
-String fullnameSync(RedditBase base) {
+String fullnameSync(RedditBaseInitializedMixin base) {
   if (base.data != null) {
     return base.data['name'];
   }
   return null;
 }
 
-void setData(RedditBase base, Map data) {
+void setData(RedditBaseInitializedMixin base, Map data) {
   base._data = data;
 }
 
-/// A base class for most DRAW objects which handles lazy-initialization of
-/// objects and Reddit API request state.
-abstract class RedditBase {
-  /// The current [Reddit] instance.
-  final Reddit reddit;
-
-  final RegExp _snakecaseRegExp = new RegExp("[A-Z]");
+abstract class RedditBaseInitializedMixin {
```

Comment:
Would you mind confirming:
I looked into the Dart Docs (https://www.dartlang.org/guides/language/language-tour), there was no reference or use of hide for exporting classes. Is this a built-in language support for the PIMPL Idiom?
(Primary reason for asking is I noticed you removed public comments.)